--- a/bcm2835-unicam.c	2025-07-24 15:12:47.695725436 +0200
+++ b/bcm2835-unicam.c	2025-07-24 15:14:03.473748523 +0200
@@ -458,7 +458,8 @@
 		.csi_dt		= MIPI_CSI2_DT_RAW14,
 		.valid_colorspaces = MASK_CS_RAW,
 	}, {
-		.fourcc		= V4L2_PIX_FMT_Y16,
+		.fourcc		= V4L2_PIX_FMT_Y16_BE,
+                .repacked_fourcc = V4L2_PIX_FMT_Y16_BE,
 		.code		= MEDIA_BUS_FMT_Y16_1X16,
 		.depth		= 16,
 		.csi_dt		= MIPI_CSI2_DT_RAW16,
